test/infra/azure/main.bicep

```diff
@@ -202,5 +205,15 @@ module deploymentScript './modules/deployment-script.bicep' = if (installKuberne
   ]
 }
 
+// Create a resource lock to prevent accidental deletion of the resource group and all its resources.
+resource deletionLock 'Microsoft.Authorization/locks@2020-05-01' = if (enableDeletionLock) {
+  scope: resourceGroup()
```

Comment:
The scope here isn't necessary since it is automatically inferred. Discussed offline with @brooke-hamilton and decided to add the resource group scope explicitly for clarity.